public/pages/DetectorDetail/containers/DetectorDetail.tsx

```diff
@@ -255,92 +256,108 @@ export const DetectorDetail = (props: DetectorDetailProps) => {
 
   return (
     <React.Fragment>
-      <EuiFlexGroup
-        direction="column"
-        style={{
-          ...(isDark
-            ? { flexGrow: 'unset' }
-            : { ...lightStyles, flexGrow: 'unset' }),
-        }}
-      >
-        <EuiFlexGroup justifyContent="spaceBetween" style={{ padding: '10px' }}>
-          <EuiFlexItem grow={false}>
-            <EuiTitle size="l">
-              <h1>
-                {detector && detector.name}{' '}
-                {detector &&
-                detector.enabled &&
-                detector.curState === DETECTOR_STATE.RUNNING ? (
-                  <EuiHealth color="success">
-                    Running since{' '}
-                    {detector.enabledTime
-                      ? moment(detector.enabledTime).format('MM/DD/YY h:mm A')
-                      : '?'}
-                  </EuiHealth>
-                ) : detector.enabled &&
-                  detector.curState === DETECTOR_STATE.INIT ? (
-                  <EuiHealth color="primary">Initializing</EuiHealth>
-                ) : detector.curState === DETECTOR_STATE.INIT_FAILURE ||
-                  detector.curState === DETECTOR_STATE.UNEXPECTED_FAILURE ? (
-                  <EuiHealth color="danger">Initialization failure</EuiHealth>
-                ) : (
-                  <EuiHealth color="subdued">
-                    {detector.featureAttributes &&
-                    detector.featureAttributes.length
-                      ? detector.disabledTime
-                        ? `Stopped at ${moment(detector.disabledTime).format(
-                            'MM/DD/YY h:mm A'
-                          )}`
-                        : 'Detector is stopped'
-                      : 'Feature required to start the detector'}
-                  </EuiHealth>
-                )}
-              </h1>
-            </EuiTitle>
-          </EuiFlexItem>
+      {!isEmpty(detector) ? (
```

Comment:
Should consider `hasError` to handle exceptions like `not found` , `timeout` etc. If any problem occur, we already have code to pop up toast:
```
 useEffect(() => {
    if (hasError) {
      toastNotifications.addDanger('Unable to find detector');
      props.history.push('/detectors');
    }
  }, [hasError]);
```